### PR TITLE
Add metadata istio overlay into bootstrap configs

### PR DESCRIPTION
**Pull request by zhenghuiwang**
part of fixing https://github.com/kubeflow/metadata/issues/101

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kubeflow/3714)

### CODE REVIEW

File: bootstrap/config/kfctl_gcp_iap.0.6.yaml [hunk 1]
```diff
@@ -62,6 +62,9 @@ spec:
     katib-ui:
     - name: overlay
       value: istio
+    metadata:
+    - name: overlay
+      value: application
```
Comment (lluunn) [inline bootstrap/config/kfctl_gcp_iap.0.6.yaml]: I thought you need overlay istio?

Comment (zhenghuiwang) [reply bootstrap/config/kfctl_gcp_iap.0.6.yaml]: Here is the manifest dir: https://github.com/kubeflow/manifests/tree/master/metadata/overlays/istio

Should this value be `istio`?

Comment (zhenghuiwang) [reply bootstrap/config/kfctl_gcp_iap.0.6.yaml]: QQ: I don' need to change any file with comments like this, right?
```
# TODO(jlewi): This file is legacy config for bootstrapper.
# we can delete it once bootstrapper is using kfctl and KfDef.
......
```

**Pull request merged**